src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -220,25 +188,14 @@ public StudentAttributes getStudentForRegistrationKey(String registrationKey) {
             // CourseStudent
             String originalKey = StringHelper.decrypt(registrationKey.trim());
             CourseStudent courseStudent = getCourseStudentEntityForRegistrationKey(originalKey);
-            if (courseStudent != null) {
-                return new StudentAttributes(courseStudent);
+            if (courseStudent == null) {
+                return null;
             }
+            return new StudentAttributes(courseStudent);
         } catch (Exception e) {
             // even if the student does not exist, it's not supposed to throw an exception
-            log.severe("Exception thrown trying to retrieve CourseStudent \n"
+            Assumption.fail("Exception thrown trying to retrieve CourseStudent \n"
```

Comment:
Yes, I remember because I'm the one who proposed that workaround. I still haven't figured out how best to handle that case until now.
(the obvious best way is to decrypt the "next URL" but that would practically invalidate all URLs that have been sent now)